src/syft/core/node/domain/client.py

```diff
@@ -65,13 +65,67 @@ def __getitem__(self, key: Union[str, int]) -> RequestMessage:
     def __repr__(self) -> str:
         return repr(self.requests)
 
-    def add_handler(self, request_handler: Dict[str, Any]) -> None:
-        self.update_handler(request_handler, keep=True)
+    def add_handler(
+        self,
+        action: str,
+        print_local: bool = False,
+        log_local: bool = False,
+        name: Optional[str] = None,
+        timeout_secs: int = -1,
+        element_quota: Optional[int] = None,
+    ) -> None:
+        handler_opts = self._validate_options(
+            action=action,
+            print_local=print_local,
+            log_local=log_local,
+            name=name,
+            timeout_secs=timeout_secs,
+            element_quota=element_quota,
+        )
 
-    def remove_handler(self, request_handler: Dict[str, Any]) -> None:
-        self.update_handler(request_handler, keep=False)
+        self._update_handler(handler_opts, keep=True)
 
-    def update_handler(self, request_handler: Dict[str, Any], keep: bool) -> None:
+    def remove_handler(
+        self,
+        action: str,
+        print_local: bool = False,
+        log_local: bool = False,
+        name: Optional[str] = None,
+        timeout_secs: int = -1,
+    ) -> None:
+        handler_opts = self._validate_options(
+            action=action,
+            print_local=print_local,
+            log_local=log_local,
+            name=name,
+            timeout_secs=timeout_secs,
+        )
+
+        self._update_handler(handler_opts, keep=False)
+
+    def _validate_options(
+        self,
+        action: str,
+        print_local: bool = False,
+        log_local: bool = False,
+        name: Optional[str] = None,
+        timeout_secs: int = -1,
+        element_quota: Optional[int] = None,
+    ) -> Dict[str, Any]:
+        handler_opts: Dict[str, Any] = {}
+        if action not in ["accept", "deny"]:
+            raise Exception("Action must be 'accept' or 'deny'")
+        handler_opts["action"] = action
+        handler_opts["print_local"] = bool(print_local)
+        handler_opts["log_local"] = bool(log_local)
+        if name is not None and name != "":
```

Comment:
MAYBE:
Simple``` if name```

Reply:
There is some weak validation before hand but you are right since we are then doing a direct comparison if there is a "truthy" value at that point its probably safe to assume we can compare it and fail it if doesnt match, actually heres a great place where we should trip and lowercase.